### DISCUSSION TITLE
How do i communicate with the server?

### DISCUSSION THREAD

**Discussion by AllNamesAreToken** [oobabooga/text-generation-webui#999] [Q&A]
Im writing a Python script, that needs to send a prompt to the server. so far i've tried

import requests

 url= "http://localhost:7860/"
    params = {"text":prompt}

and

response = requests.post("http://localhost:7860", json={"text": prompt})

The best i get is an 405 error, that means it does not understand what i want :/ 
Is there some kind of way to get it to take the prompt and read the answer? 

Thanks!

**Comment by bmoconno** [comment]
You're missing a lot of key parts that are required, if you haven't already, you should check out the [api example](https://github.com/oobabooga/text-generation-webui/blob/main/api-example.py).

**Reply by AllNamesAreToken** [reply]
even with this it spits out a 

Traceback (most recent call last):
  File "D:\Programieren\pythonProject1\Command.py", line 51, in <module>
    reply = response["data"][0]
KeyError: 'data'

Error, ive set up the server to listen and --no-stream. 
i can use the web ui on all my local devices.  

do i have to configure an seperate json file?
:/

**Reply by bmoconno** [reply]
Are you running the server in `default` or `notebook`? That's currently the only way to use the API.

**Reply by AllNamesAreToken** [reply]
We are getting some where :) it took  the prompt in default mode and spat out an answer. now i just have to find out how to tweak it to not talk as 2 persons :) but i guess i can manage that :) 

Thank you very much

**Reply by bmoconno** [reply]
Check out this discussion #656, there's some people who are working on that using `default` mode and I just yesterday made a branch to allow using the API for chat mode. It's not 100% yet, but it does work if you want to test it out.
